superset/migrations/shared/migrate_viz/base.py

```diff
@@ -68,22 +69,50 @@ def _migrate(self) -> None:
 
             if key in self.rename_keys:
                 rv_data[self.rename_keys[key]] = value
+                continue
 
             if key in self.remove_keys:
                 continue
 
             rv_data[key] = value
 
+        if is_feature_enabled("GENERIC_CHART_AXES"):
+            self._migrate_temporal_filter(rv_data)
+
         self.data = rv_data
 
     def _post_action(self) -> None:
-        """some actions after migrate"""
+        """Some actions after migrate"""
+
+    def _migrate_temporal_filter(self, rv_data: Dict[str, Any]) -> None:
+        """Adds a temporal filter."""
+        granularity_sqla = rv_data.pop("granularity_sqla", None)
+        time_range = rv_data.pop("time_range", conf.get("DEFAULT_TIME_FILTER"))
```

Comment:
I believe there's the risk that the property is defined with a `None` value, in which case the default value wouldn't kick in (I've learned to be sceptical about form data quality 😄 ). So I think it may be a good idea to do
```suggestion
        time_range = rv_data.pop("time_range", None) or conf.get("DEFAULT_TIME_FILTER")
```